### PR TITLE
Add setter on query builder

### PR DESCRIPTION
**Pull request by olamothe**
TLDR: 

Salesforce Locker service modifies underscore.js module to add `strict mode`;
This means the _.extend function starts throwing when extending object with getter but no setter...

https://coveord.atlassian.net/browse/JSUI-2640

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

File: src/controllers/QueryController.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import { history } from 'coveo.analytics';
+import { NullStorage } from 'coveo.analytics/dist/storage';
```
Comment (jeremierobert-coveo) [inline src/controllers/QueryController.ts]: Change on this file are required ?

File: src/ui/Base/QueryBuilder.ts [hunk 2]
```diff
@@ -528,11 +528,19 @@ export class QueryBuilder {
...
+
   private get facets() {
     if (Utils.isEmptyArray(this.facetRequests)) {
       return undefined;
     }
 
     return this.facetRequests;
   }
+
+  private set facets(facets: IFacetRequest[]) {
```
Comment (jeremierobert-coveo) [inline src/ui/Base/QueryBuilder.ts]: Is this class the only case of a class with only getter ?

**Comment by olamothe** [comment]
@jeremierobert-coveo No, there's a ton of other place where we use setter.

The fact that locker service is modifying the code (because, yes, adding "use strict" means the code is not similar to what the author of the underscore library intended), is very problematic.

We need to think of a better solution to test this..

**Comment by jeremierobert-coveo** [comment]
I'm not denying the fact that what Locker Service does is not problematic 😄 .

Do you think the salesforce integration should add some E2E to test client possible customization in Lightning ?

**Pull request merged**